_CreateCompileItemsForOpenApiReferences adds generated .cs to Compile even if they are already there.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When using `OpenApiReference` and `OpenApiCodeDirectory` is set to a project folder, the generated C# file is added by the SDK and is again added by [_CreateCompileItemsForOpenApiReferences](https://github.com/dotnet/aspnetcore/blob/ba3c78b4fbb387d8785dbfb762128a767b9c392c/src/Tools/Extensions.ApiDescription.Client/src/build/Microsoft.Extensions.ApiDescription.Client.targets#L121) causing:

```
CSC : warning CS2002: Source file 'MyApi.generated.cs' specified multiple times
```

### Expected Behavior

The file is not added if it is already in the `Compile` items.

### Steps To Reproduce

Set a code directory on the csproj file:

```xml
  <PropertyGroup>
    <OpenApiCodeDirectory>$(MSBuildProjectDirectory)</OpenApiCodeDirectory>
  </PropertyGroup>
```

### Exceptions (if any)

_No response_

### .NET Version

.NET SDK 6.0.415

### Anything else?

_No response_

--- a/src/Tools/Extensions.ApiDescription.Client/src/build/Microsoft.Extensions.ApiDescription.Client.targets
+++ b/src/Tools/Extensions.ApiDescription.Client/src/build/Microsoft.Extensions.ApiDescription.Client.targets
@@ -118,8 +118,8 @@
           Exclude="@(TypeScriptCompile)"
           Condition="'%(_Files.Extension)' == '.ts' OR '%(_Files.Extension)' == '.tsx'" />
 
-      <Compile Include="@(_Files)"
-          Exclude="@(Compile)"
+      <Compile Include="@(_Files->'%(FullPath)')"
+          Exclude="@(Compile->'%(FullPath)')"
           Condition="'$(DefaultLanguageSourceExtension)' != '.ts' AND '%(_Files.Extension)' == '$(DefaultLanguageSourceExtension)'" />
     </ItemGroup>
 
@@ -138,7 +138,7 @@
       <TypeScriptCompile Include="$(_TypeScriptCompileItemsFromDirectories)" Exclude="@(TypeScriptCompile)" />
 
       <Compile Include="$(_CompileItemsFromDirectories)"
-          Exclude="@(Compile)"
+          Exclude="@(Compile->'%(FullPath)')"
           Condition="'$(DefaultLanguageSourceExtension)' != '.ts'" />
 
       <FileWrites Include="@(_Files);$(_TypeScriptCompileItemsFromDirectories);$(_CompileItemsFromDirectories)"

